frontend/src/store/slices/promptSlice.ts

```diff
@@ -0,0 +1,124 @@
+/* ---------------------------------------------------- */
+/* Luis, I still have zero idea what I am doing, hahaha */
+/* ---------------------------------------------------- */
+
+import {
+  createAsyncThunk,
+  createSlice,
+  PayloadAction,
+} from '@reduxjs/toolkit';
+
+// Temporarily left here and might be modified later
+interface PromptData {
+  role: string;
+  context: string;
+  task: string;
+  output: string;
+  constraints: string;
+  language: string;
+}
+
+interface ApiResponse {
+  userId?: string;
+  promptId: string;
+  content: string;
+  metadata: {
+    language: string;
+    model: string;
+    formattedPromptPreview: string;
+  };
+}
+
+interface ApiErrorResponse {
+  error: string;
+}
+
+interface PromptState {
+  output: string | null;
+  loading: 'idle' | 'pending' | 'succeeded' | 'failed';
+  error: string | null;
+}
+
+const initialState: PromptState = {
+  output: null,
+  loading: 'idle',
+  error: null,
+};
+
+const API_BASE_URL = import.meta.env.VITE_API_BASE_URL;
+
+if (!API_BASE_URL) {
+  console.error('Error: VITE_API_BASE_URL is not defined in the environment variables.');
+}
+
+export const createPrompt = createAsyncThunk<
+  string,
+  PromptData,
+  { rejectValue: string }
+>('prompts/createPrompt', async (promptData: PromptData, { rejectWithValue }) => {
+  if (!API_BASE_URL) {
+    return rejectWithValue('Base URL is not configured.');
+  }
+  try {
+    const response = await fetch(`${API_BASE_URL}/prompts`, {
```

Comment:
the API call, should be in the /api folder something like this /api/prompts and then we can import it in the slices.